core-rust/core-api-server/src/core_api/conversions/pagination.rs

```diff
@@ -0,0 +1,83 @@
+use radix_engine::types::*;
+
+use crate::core_api::*;
+
+pub struct SizeRange {
+    pub min: usize,
+    pub default: usize,
+    pub max: usize,
+}
+
+pub fn extract_valid_size(
+    provided: Option<i32>,
+    size_range: SizeRange,
+) -> Result<usize, ExtractionError> {
+    let SizeRange { min, default, max } = size_range;
+    match provided {
+        Some(provided) => {
+            if provided < 0 {
+                None
+            } else {
+                let provided = provided as usize;
+                if provided < min || provided > max {
+                    None
+                } else {
+                    Some(provided)
+                }
+            }
+        }
+        None => Some(default),
+    }
+    .ok_or(ExtractionError::InvalidSize { min, max })
+}
+
+pub fn extract_continuation_token<T: ScryptoDecode>(
+    continuation_token: Option<String>,
+) -> Result<Option<T>, ExtractionError> {
```

Comment:
this kind of functions become more general-purpose when you drop both the input and output `Option<>`.
(i.e. `extract_continuation_token(String) -> Result<T, E>`)

it is typically easy enough for the caller to do `option.map(extract_continuation_token)` anyways.

Reply:
Hmm so I want to build a toolkit which mandates/encourages correct use of the new API specs. In that case, continuation token is always optional; so I think this is correct. Will add a comment 👍